tests/lib/Config/ItemType/IncludedItemTypeResolverTest.php

```diff
@@ -0,0 +1,165 @@
+     * @param array<string> $inputItemTypes
+     */
+    public function testResolve(
+        array $expectedItemTypes,
+        array $inputItemTypes,
+        bool $useLogger,
+        ?string $siteAccess = null
+    ): void {
+        $this->configureConfigResolverToReturnIncludedContentTypes($siteAccess);
+
+        if ($useLogger) {
+            $this->configureLoggerToLogWarning($expectedItemTypes, $inputItemTypes);
+        }
+
+        self::assertEquals(
+            $expectedItemTypes,
+            $this->itemTypeResolver->resolve($inputItemTypes, $useLogger, $siteAccess)
+        );
+    }
+
+    /**
+     * @phpstan-return iterable<array{
+     *  array<string>,
+     *  array<string>,
+     *  bool,
+     *  3?: string
+     * }>
+     */
+    public function provideDataForTestResolve(): iterable
+    {
+        yield [
+            ['d_foo', 'd_bar', 'd_baz'],
+            ['d_foo', 'd_bar', 'd_baz'],
+            false,
+        ];
+
+        yield [
+            ['f_foo', 'f_bar', 'f_baz'],
+            ['f_foo', 'f_bar', 'f_baz'],
+            false,
+            self::SITE_ACCESS_FOO,
+        ];
+
+        yield [
+            ['b_foo', 'b_bar', 'b_baz'],
+            ['b_foo', 'b_bar', 'b_baz'],
+            false,
+            self::SITE_ACCESS_BAR,
+        ];
+
+        yield [
+            ['d_foo'],
+            ['d_foo'],
+            false,
+        ];
+
+        yield [
+            ['d_foo'],
+            ['d_foo', 'test', 'folder'],
+            true,
+        ];
+
+        yield [
+            [],
+            ['test', 'folder'],
+            true,
+        ];
+
+        yield [
+            [],
+            ['foo', 'bar'],
+            true,
+        ];
+    }
+
+    private function configureConfigResolverToReturnIncludedContentTypes(?string $siteAccess = null): void
+    {
+        $configuredItemTypes = [];
+
+        if (null === $siteAccess) {
+            $configuredItemTypes = self::CONFIGURED_ITEM_TYPES_DEFAULT;
+        }
+
+        if (isset(self::SITE_ACCESS_MAP[$siteAccess])) {
+            $configuredItemTypes = self::SITE_ACCESS_MAP[$siteAccess];
+        }
+
+        $this->configResolver
+            ->expects(self::once())
+            ->method('getParameter')
+            ->with('included_content_types', 'ezrecommendation', $siteAccess)
+            ->willReturn($configuredItemTypes);
+    }
+
+    /**
+     * @param array<string> $expectedItemTypes
+     * @param array<string> $inputItemTypes
+     */
+    private function configureLoggerToLogWarning(array $expectedItemTypes, array $inputItemTypes): void
+    {
+        $message = sprintf(
+            'Item types: %s are not configured as included item types'
+            . ' and have been removed from resolving criteria',
+            implode(', ', array_diff($inputItemTypes, $expectedItemTypes))
+        );
+
+        $this->logger
+            ->expects(self::once())
+            ->method('warning')
+            ->with($message);
```

Comment:
If we do not care about the array diff that presents the removed items, this could be replaced with 
```suggestion
            ->with(self::matches($message));
```
where `$message`:
```php
$message = 'Item types: %s are not configured...';
```
since `self::matches()` does a "sprintf-like" regexp conversion (see \PHPUnit\Framework\Constraint\StringMatchesFormatDescription::createPatternFromFormat).

(Passing value which is not an assertion is silently converted to `self::equals`)

This would make mock configuration a bit easier IMO.